app/src/features/datasets/components/AttachmentItemMenuButton.tsx

```diff
@@ -48,66 +49,65 @@ const AttachmentItemMenuButton: React.FC<IAttachmentItemMenuButtonProps> = (prop
-                  <Icon path={mdiTrayArrowDown} size={0.875} />
-                </ListItemIcon>
-                Download Document
-              </MenuItem>
-            ) : (
-              <MenuItem
-                onClick={() => {
-                  props.onRequestAccess(props.artifact);
-                  handleClose();
-                }}
-                data-testid="attachment-action-menu-request-access">
-                <ListItemIcon>
-                  <Icon path={mdiLockOutline} size={0.875} />
-                </ListItemIcon>
-                Request Access
-              </MenuItem>
-            )}
-
-            {props.hasAdministrativePermissions && (
-              <>
+            onClose={handleClose}>
+            <MenuList aria-labelledby={`attachment-item-menu-button=${props.artifact.artifact_id}`}>
+              {canDownload ? (
                 <MenuItem
                   onClick={() => {
-                    props.onApplySecurity(props.artifact);
-                    setAnchorEl(null);
+                    props.onDownload(props.artifact);
+                    handleClose();
                   }}
                   data-testid="attachment-action-menu-download">
                   <ListItemIcon>
-                    <Icon path={mdiLockPlus} size={0.875} />
+                    <Icon path={mdiTrayArrowDown} size={0.875} />
                   </ListItemIcon>
-                  Apply Security to Document
+                  Download Document
                 </MenuItem>
-
+              ) : (
                 <MenuItem
                   onClick={() => {
-                    console.log('Delete artifact not implemented yet.');
+                    props.onRequestAccess(props.artifact);
                     handleClose();
                   }}
-                  data-testid="attachment-action-menu-delete">
+                  data-testid="attachment-action-menu-request-access">
                   <ListItemIcon>
-                    <Icon path={mdiTrashCanOutline} size={0.8} />
+                    <Icon path={mdiLockOutline} size={0.875} />
                   </ListItemIcon>
-                  Delete Document
+                  Request Access
                 </MenuItem>
-              </>
-            )}
+              )}
+
+              {props.hasAdministrativePermissions && (
+                <>
+                  <MenuItem
+                    onClick={() => {
+                      props.onApplySecurity(props.artifact);
+                      setAnchorEl(null);
+                    }}
+                    data-testid="attachment-action-menu-download">
+                    <ListItemIcon>
+                      <Icon path={mdiLockPlus} size={0.875} />
+                    </ListItemIcon>
+                    Apply Security to Document
+                  </MenuItem>
+
+                  <MenuItem
+                    onClick={() => {
+                      console.log('Delete artifact not implemented yet.');
```

Comment:
nit

Reply:
This gets addressed in https://github.com/bcgov/biohubbc-platform/pull/203/files#diff-12fa4d97ebcdda177e6b9872e35f0e2fc2b2b0e8e57955e64920b86dcb302e75